lib/IRGen/GenDecl.cpp

```diff
@@ -488,8 +488,13 @@ void IRGenModule::emitSourceFile(SourceFile &SF) {
       this->addLinkLibrary(LinkLibrary("stdc++", LibraryKind::Library));
 
     // Do not try to link Cxx with itself.
-    if (!getSwiftModule()->getName().is("Cxx"))
-      this->addLinkLibrary(LinkLibrary("swiftCxx", LibraryKind::Library));
+    if (!getSwiftModule()->getName().is("Cxx")) {
+      bool isStatic = Context.getModuleByName("Cxx")->isStaticLibrary();
```

Comment:
I suspect the crash is here because the `Cxx` module is not known, and `Context.getModuleByName("Cxx")` is returning NULL.

Reply:
I suppose that if the module is not found, we can assume that we are dynamically linking.